dotCMS/src/main/java/com/dotcms/contenttype/model/field/FieldTypeFactory.java

```diff
@@ -0,0 +1,74 @@
+package com.dotcms.contenttype.model.field;
+
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.util.UtilMethods;
+import com.liferay.portal.language.LanguageException;
+import com.liferay.portal.language.LanguageUtil;
+import com.liferay.portal.model.User;
+
+import java.util.*;
+
+/**
+ * Created by freddyrodriguez on 7/27/17.
+ */
+public class FieldTypeFactory {
+
+    private Map<String, Field> fieldsMap;
+
+    private static class SingletonHolder {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add a private constructor to hide the implicit public one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1118)